src/components/pages/Profile.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState, useRef, useEffect, ChangeEvent } from "react";
+import Header from "./Header";
+import { UserAuth } from "../../context/UserAuthContext";
+import { ref, uploadBytes, getDownloadURL } from "firebase/storage";
+import { storage } from "../../firebase/firebase";
+
+const Profile = () => {
+    const { user, profileData } = UserAuth();
+    const [avatarURL, setAvatarURL] = useState("");
```

Comment:
Keep this in constants file
     const defaultAvatarURL =   "https://firebasestorage.googleapis.com/v0/b/post-project-80c0a.appspot.com/o/images%2FUserAvatar.png?alt=media&token=af9525b4-1f0e-468f-9352-f04004248dfc";

```suggestion
    const [avatarURL, setAvatarURL] = useState(defaultAvatarURL);
```